File: tester/harmony/rnoh/src/main/ets/RNOHCorePackage/turboModules/AccessibilityInfoTurboModule.ts
import accessibility from '@ohos.accessibility';
import type { TurboModuleContext } from '../../RNOH/TurboModule';
import { TurboModule } from '../../RNOH/TurboModule';

export class AccessibilityInfoTurboModule extends TurboModule {
  public static readonly NAME = 'AccessibilityInfo';

  constructor(ctx: TurboModuleContext) {
    super(ctx);
  }

  public isTouchExplorationEnabled(
    resolveCallback: (isScreenReaderEnabled: boolean) => void,
  ) {
    const isScreenReaderEnabled = accessibility.isOpenTouchGuideSync();
    resolveCallback(isScreenReaderEnabled);
  }
}

File: react-native-harmony/harmony/cpp/RNOHCorePackage/TurboModules/AccessibilityInfoTurboModule.cpp
#include "RNOHCorePackage/TurboModules/AccessibilityInfoTurboModule.h"

namespace rnoh {
using namespace facebook;

AccessibilityInfoTurboModule::AccessibilityInfoTurboModule(
    const ArkTSTurboModule::Context ctx,
    const std::string name)
    : ArkTSTurboModule(ctx, name) {
  methodMap_ = {ARK_METHOD_METADATA(isTouchExplorationEnabled, 1)};
}

} // namespace rnoh
